velox/vector/tests/utils/VectorMaker.h

```diff
@@ -588,21 +588,82 @@ class VectorMaker {
         arrays.push_back(std::nullopt);
         continue;
       }
-
       std::vector<std::optional<T>> elements;
-      for (const auto& element : arrayObject) {
-        if (element.isNull()) {
-          // Null element.
-          elements.push_back(std::nullopt);
-        } else {
-          elements.push_back(detail::jsonValue<T>(element));
-        }
-      }
+      elementVectorFromJsonHelper(elements, arrayObject);
+    }
+    return arrayVectorNullable<T>(arrays, arrayType);
+  }
 
-      arrays.push_back(elements);
+  template <typename T>
+  void elementVectorFromJsonHelper(
```

Comment:
we tend not to name methods xxxHelper; rather use names that start with a verb

appendElementsFromJsonArray(arrayObject, elements)